controllers/provider-aws/pkg/webhook/controlplane/mutator.go

```diff
@@ -156,3 +168,82 @@ func ensureVolumes(ps *corev1.PodSpec) {
 	ps.Volumes = controlplane.EnsureVolumeWithName(ps.Volumes, cloudProviderConfigVolume)
 	ps.Volumes = controlplane.EnsureVolumeWithName(ps.Volumes, cloudProviderSecretVolume)
 }
+
+func (m *mutator) mutateOperatingSystemConfig(osc *extensionsv1alpha1.OperatingSystemConfig) error {
+	// Mutate kubelet.service unit, if present
+	if u := controlplane.UnitWithName(osc.Spec.Units, "kubelet.service"); u != nil && u.Content != nil {
+		if err := m.ensureKubeletServiceUnitContent(u.Content); err != nil {
+			return err
+		}
+	}
+
+	// Mutate kubelet configuration file, if present
+	if f := controlplane.FileWithPath(osc.Spec.Files, "/var/lib/kubelet/config/kubelet"); f != nil && f.Content.Inline != nil {
+		if err := m.ensureKubeletConfigFileContent(f.Content.Inline); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *mutator) ensureKubeletServiceUnitContent(content *string) error {
+	var opts []*unit.UnitOption
+	var err error
+
+	// Deserialize unit options
+	if opts, err = m.unitSerializer.Deserialize(*content); err != nil {
+		return errors.Wrap(err, "could not deserialize kubelet.service unit content")
+	}
+
+	ensureKubeletServiceUnitOptions(opts)
+
+	// Serialize unit options
+	if *content, err = m.unitSerializer.Serialize(opts); err != nil {
+		return errors.Wrap(err, "could not serialize kubelet.service unit options")
+	}
+
+	return nil
+}
+
+func (m *mutator) ensureKubeletConfigFileContent(fci *extensionsv1alpha1.FileContentInline) error {
+	// Decode kubelet configuration from inline content
+	kubeletConfig := &kubeletconfigv1beta1.KubeletConfiguration{}
+	if err := m.kubeletConfigCodec.Decode(fci, kubeletConfig); err != nil {
+		return errors.Wrap(err, "could not decode kubelet configuration")
+	}
+
+	ensureKubeletConfiguration(kubeletConfig)
+
+	// Encode kubelet configuration into inline content
+	if err := m.kubeletConfigCodec.Encode(kubeletConfig, fci); err != nil {
+		return errors.Wrap(err, "could not encode kubelet configuration")
+	}
+
+	return nil
+}
+
+func ensureKubeletServiceUnitOptions(opts []*unit.UnitOption) {
+	if opt := controlplane.UnitOptionWithSectionAndName(opts, "Service", "ExecStart"); opt != nil {
+		command := controlplane.DeserializeCommandLine(opt.Value)
+		command = ensureKubeletCommandLineArgs(command)
+		opt.Value = controlplane.SerializeCommandLine(command, 1, " \\\n    ")
+	}
+}
+
+func ensureKubeletCommandLineArgs(command []string) []string {
+	command = controlplane.EnsureStringWithPrefix(command, "--cloud-provider=", "aws")
+	command = controlplane.EnsureNoStringWithPrefix(command, "--provider-id=")
+	command = controlplane.EnsureNoStringWithPrefix(command, "--enable-controller-attach-detach=")
```

Comment:
This is not needed I guess - Gardener won't add this flag.

Reply:
See above, this one is related to CSI.

Reply:
Removed as initially suggested.